manifests/extension.pp

```diff
@@ -110,23 +115,36 @@
 
   $lowercase_title = downcase($title)
 
+  # Ensure "<extension>." prefix is present in setting keys if requested
+  if $settings_prefix {
+    $full_settings = ensure_prefix($settings, "${name}.")
```

Comment:
This should probably be configurable optionally for cases like the one mentioned above. Using the `$lowercase_title` is probably desired to.